### DISCUSSION TITLE
Is it possible to model something like this with pyg or pygTemporal?

### DISCUSSION THREAD

**Discussion by DogitoErgoSum** [pyg-team/pytorch_geometric#4324] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/51344466/159496296-37026629-53eb-4e43-9742-54cef54a7e6a.png)

Where a circle is a node and different colors represent different node types.

**Comment by rusty1s** [comment, marked as answer]
As outlined in https://github.com/pyg-team/pytorch_geometric/discussions/4323, this can be achieved by a list of events:
```
(src=0, dst=1, time=t1)
(src=0, dst=2, time=t2)
(src=1, dst=2, time=t2)
```

**Reply by DogitoErgoSum** [reply]
i am not following.
As an example, if orange type has a feature size of 5 and green a feature size of 3, i m not sure how would one define what type they belong or its feature size with the above, like one does with heterodata.

**Reply by rusty1s** [reply]
Yes, the feature set of different types/messages currently need to have the same dimensionality in `TemporalData` (or you will need to work with padding features to same length). Sorry :(

**Reply by DogitoErgoSum** [reply]
Ty for your answer. I will model the sequence has one heterogeneous graph with time has as a feature.

**Reply by rusty1s** [reply]
That should work as well :)
